internal/pkg/deploy/cloudformation/project.go

```diff
@@ -449,48 +449,73 @@ func (cf CloudFormation) waitForStackSetOperation(stackSetName, operationID stri
 }
 
 // DeleteProject deletes all project specific StackSet and Stack resources.
-func (cf CloudFormation) DeleteProject(name string, accounts, regions []string) error {
-	stackSetName := fmt.Sprintf("%s-infrastructure", name)
+func (cf CloudFormation) DeleteProject(projectName string) error {
+	stackSetName := fmt.Sprintf("%s-infrastructure", projectName)
+	if err := cf.deleteProjectStackSet(stackSetName); err != nil {
+		return err
+	}
+	return cf.delete(fmt.Sprintf("%s-infrastructure-roles", projectName))
+}
 
-	if _, err := cf.client.DeleteStackInstances(&cloudformation.DeleteStackInstancesInput{
-		Accounts:     aws.StringSlice(accounts),
-		RetainStacks: aws.Bool(false),
-		Regions:      aws.StringSlice(regions),
+func (cf CloudFormation) deleteProjectStackSet(stackSetName string) error {
+	stackInstances, err := cf.client.ListStackInstances(&cloudformation.ListStackInstancesInput{
 		StackSetName: aws.String(stackSetName),
-	}); err != nil {
-		return fmt.Errorf("DeleteStackInstances for stackset %s, accounts %s, and regions %s: %w",
-			stackSetName, accounts, regions, err)
+	})
+
+	if err != nil {
+		// If the stackset doesn't exist - just move on.
+		if stackSetDoesNotExist(err) {
+			return nil
+		}
+		return fmt.Errorf("fetching existing project stack instances: %w", err)
+	}
+
+	// We want to delete all the stack instances, so we create
+	// a set of account ids and regions.
+	accountSet := map[string]bool{}
+	regionSet := map[string]bool{}
```

Comment:
nit: not important but this just startled me as I don't think this is the common way of initializing maps although I think it works:
> To initialize a map, use the built in make function:

https://blog.golang.org/go-maps-in-action

Reply:
`malloc(sizeof(map[string]string))` kept failing, so I went with this. hehe